### ISSUE TITLE
Replace current recyclerview implementation with BindableAdapter usage

### ISSUE THREAD

**Issue by rt4914** [oppia/oppia-android#632]
Currently the entire project has two types of RecyclerView implementation:
1. Recyclerview using BindableAdapter (OngoingStoryListActivity, CompletedStoryListActivity, etc.) uses it)

2. Recyclerview with the RecyclerViewAdapter which inflates multiple layouts.

We need to change the implementation such that all RecyclerViewAdapter to  use BindableAdapter (i.e. approach similar to OngoingStoryListActivity, CompletedStoryListActivity).

## Checklist
- [x] `ContinuePlaying` (assigned to @veena14cs )
- [x] `HomeFragment` (assigned to @jcqli)
- [x] `StoryActivity` (Already done)
- [x] `TopicReview/Revision` (Assigned to @Sarthak2601 )
- [x] `TopicPlay/Lessons`
- [x] `TopicTrain/Practice` (assigned to @Luffy18346 )
- [x] `HelpActivity` (Assigned to @SayantanBanerjee16 )
- [x] `HintsAndSolutionAdapter`  (Assigned to @anandwana001 )
- [x] `OngoingListAdapter`  (open)
- [x] `TopicListAdapter` (assigned to @jcqli)
- [x] `LanguageSelectionAdapter` (assigned to @FareesHussain)
- [x] `StorySummaryAdapter`  (open)
- [ ] `PromotedStoryListAdapter` (assigned to @masclot )

**Comment by veena14cs** [comment]
@rt4914 I will be taking up Continue Playing task in this list.

**Comment by veena14cs** [comment]
@rt4914  StoryActivity already has implementation of Bindable adapter.

**Comment by NullByte08** [comment]
@rt4914 Can I work on the `HomeFragment` task? I will replace both `TopicListAdapter` and `PromotedStoryListAdapter`.

**Comment by rt4914** [comment]
> @rt4914 Can I work on the `HomeFragment` task? I will replace both `TopicListAdapter` and `PromotedStoryListAdapter`.

@NullByte08 I would suggest you to work on simpler one first as HomeFragment is the hardest one. Still its completely fine if you work on that.

**Comment by Sarthak2601** [comment]
@rt4914 Can I work on `TopicReview/Revision` task ?

**Comment by rt4914** [comment]
> @rt4914 Can I work on `TopicReview/Revision` task ?

Sure.

**Comment by NullByte08** [comment]
@rt4914 Can I work on `FAQMain` task then?

**Comment by SayantanBanerjee16** [comment]
@rt4914 I think `HelpActivity` should also come in this list. Can you please add it and assign me `HelpActivity`?

**Comment by rt4914** [comment]
@SayantanBanerjee16 @NullByte08 
The FAQMain is actually `HelpActivity` only. Now considering that @SayantanBanerjee16 has already worked on `HelpActivity` it would be nice to assign it to him. And also, as @NullByte08 had mentioned earlier regarding `HomeActivity` but I do think it is hard also, but as none of the easy one's are left now @NullByte08 can work on `HomeActivity` .

These arrangements have been done keep in mind the PRs that  you have worked on in past.

**Comment by NullByte08** [comment]
@rt4914 Alright sure.

**Comment by abhinavraj23** [comment]
@rt4914 can I work on TopicPlay/Lessons?

**Comment by rt4914** [comment]
@abhinavraj23 @NullByte08 @Sarthak2601 @SayantanBanerjee16  
NOTE for all.
The files that you are working on will also get used by other contributor on other PRs too, so you can expect merge conflicts if PR takes lot of time. Also, these are fairly difficult at the same time very good for knowledge tasks. So if you get stuck on any of these then you can leave a particular issue if you want, its completely fine.

**Comment by rt4914** [comment]
@NullByte08 and @abhinavraj23 Any updates in this? If this is not doable its fine, we can unassign you from this.

**Comment by NullByte08** [comment]
@rt4914 I am facing a lot of errors in the `HomeFragment` since it has two recyclerviews. Can I take some more time? I will address the issues to you that I am facing if I still don't get a solution for them.

**Comment by rt4914** [comment]
> @rt4914 I am facing a lot of errors in the `HomeFragment` since it has two recyclerviews. Can I take some more time? I will address the issues to you that I am facing if I still don't get a solution for them.

@rt4914 Yeah sure, take your time.

**Comment by jcqli** [comment]
@rt4914 Was `TopicPlay/Lessons` ever done? Happy to work on it!

**Comment by BenHenning** [comment]
``TopicListAdapter`` still needs to be removed/migrated, so that's open. Assigning you @jcqli.

**Comment by FareesHussain** [comment]
I'm working on `StorySummaryAdapter`

**Comment by prayutsu** [comment]
@rt4914 I am taking `OngoingListAdapter`

**Comment by rt4914** [comment]
@prayutsu Suggest creating a PR once #2320 is merged as it does some major changes to `Recently Played Activity`

**Comment by prayutsu** [comment]
@rt4914 Thanks, will wait till it gets merged.

**Comment by imraghav20** [comment]
@rt4914 Can I work on `HintsAndSolutionsAdapter` ?
Or you can suggest me something easier if not assigned to anyone. Otherwise I will be happy working on `HintsAndSolutionsAdapter`.

**Comment by rt4914** [comment]
@imraghav20 Definitely have a go at it. It should be doable considering there are enough code samples already in codebase. But if you are not able to solve it thats fine, just mention it and we will assign you something else.

**Comment by MaskedCarrot** [comment]
may I take up `HintsAndSolutionAdapter`? 
@rt4914  @imraghav20

**Comment by imraghav20** [comment]
> may I take up `HintsAndSolutionAdapter`?
> @rt4914 @imraghav20

@MaskedCarrot I am currently working on it.

**Comment by MaskedCarrot** [comment]
@rt4914 if TopicPlay/Lessons is still open I would like to work on it

**Comment by FareesHussain** [comment]
> @rt4914 if TopicPlay/Lessons is still open I would like to work on it

I think TopicPlay/Lessons is the same as StorySummaryAdapter. I'm working on StorySummaryAdapter (https://github.com/oppia/oppia-android/pull/2447) now.

**Comment by anandwana001** [comment]
@imraghav20 any update?

**Comment by imraghav20** [comment]
> @imraghav20 any update?

I have figured out the code and where are the necessary changes to be made. My exams have started from today (8 Feb) and will continue till 14 Feb. So, I might take bit more time.

**Comment by jonathanalvares9009** [comment]
@FareesHussain can you assign me StorySummaryAdapter

**Comment by FareesHussain** [comment]
> @FareesHussain can you assign me StorySummaryAdapter

@anandwana001 PTAL.
@jonathanalvares9009 you can start working on this

**Comment by bhaktideshmukh** [comment]
Hey @anandwana001 currently if you are not working on MyDownloadsViewPagerAdapter can I take this up?

**Comment by anandwana001** [comment]
> Hey @anandwana001 currently if you are not working on MyDownloadsViewPagerAdapter can I take this up?

@bhaktideshmukh Sure, also assign yourself in description

**Comment by chetanmuliya** [comment]
can I work on this -> OngoingListAdapter ?

**Comment by ArchitJain1201** [comment]
@rt4914 Can I work on `StorySummaryAdapter` ?

**Comment by masclot** [comment]
OngoingListAdapter was renamed PromotedStoryListAdapter in #4704
I'm going to start working on this one.

**Comment by masclot** [comment]
There is only one remaining class in the codebase that implements RecyclerView.Adapter: FragmentStateAdapter
This in turn is extended by two more classes: ViewPagerAdapter and MyDownloadsViewPagerAdapter

I don't know if it makes sense to refactor those. Otherwise this bug can be closed.

**Comment by adhiamboperes** [comment]
Unassigning masclot since his PR has been merged.
I think this needs re-triage as per @masclot's comment.

**Comment by BenHenning** [comment]
Agreed--thanks @masclot & @adhiamboperes! I think we can go ahead and close this.

**Comment by adhiamboperes** [comment]
Reopening this since we are reverting the associated pull request.
